src/components/channelshelf/channelshelf.html

```diff
@@ -22,8 +22,8 @@
         }"
         field-def="encoding[channelId]"
         show-type="true"
-        show-caret="true"
-        disable-caret="encoding[channelId].aggregate==='count'"
+        show-caret="hasFunctions"
```

Comment:
If I understand correctly: when you turn show-caret to false, it will cause incorrect layout.  (The field will get shifted due to missing caret.) 

I think just applying disable-caret in the line below is sufficient.